sw/device/rom_exts/manifest.hjson

```diff
@@ -0,0 +1,156 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  fields : [
+    { name: "manifest_identifier",
+      desc: '''ROM_EXT Manifest Identifier.
+
+            TODO
+            ''',
+      size: "32",
+    },
+    { name: "reserved",
```

Comment:
Using a specific name seems brittle here.

You probably have two choices for making this nicer:
- a `type` field that is either "field" or "reserved", and if it's a field, you generate for it, otherwise you don't. And if you don't generate anything for it, the lack of a name field isn't a problem.
- choosing a design like reggen does with `{skip_to: <offset>}` to cover the reserved areas, rather than `{name:"reserved", size:…}`

Later, we should absolutely be checking the format of this data, but I don't know if we know that yet.

Reply:
It could be done in a follow-up, it might be worth having something in that spits out `manifest.h` with the appropriate dfines.

Reply:
What do you think?

Reply:
I don't know what you mean - this whole script spits out the manifest.h with the appropriate defines.

Reply:
I mean that the format might be brittle but it works, so would it be reasonable to merge this (when other issues are resolved), and change the `.hjson` format after (will involve changes to the generator).

Reply:
The first of those alternatives is a simple change, and makes it a lot less brittle - let's go with that. The project has learned a lot from defining these data formats, it seems pointless to go through the process of learning it all again.

Reply:
I agree with @lenary, using an internally tagged representation is better than overloading the meaning of `name`. This would also enable, if we choose to in the future, defining groups of fields, e.g. signed area, extension area, etc., which would simplify code that works groups of fields rather than individual fields.

Reply:
Sounds good, I will go with `type` instead of `skipto`, as the way I calculate the offset is from the size of the fields rather than skipping to a specific  address.

Reply:
Done.